### PR TITLE
Extracted generic render pass for a cookie renderer and picker to a new modules

### PR DESCRIPTION
**Pull request by mvaligursky**
- just a refactor on a way to standalone render passes

### CODE REVIEW

File: src/framework/graphics/picker.js [hunk 1]
```diff
@@ -4,19 +4,16 @@ import { ADDRESS_CLAMP_TO_EDGE, FILTER_NEAREST, PIXELFORMAT_RGBA8 } from '../../
...
 import { getApplication } from '../globals.js';
-import { Debug, DebugHelper } from '../../core/debug.js';
-import { BlendState } from '../../platform/graphics/blend-state.js';
-import { RenderPass } from '../../platform/graphics/render-pass.js';
+import { Debug } from '../../core/debug.js';
+import { RenderPassPicker } from './render-pass-picker.js';
 
 const tempSet = new Set();
-const tempMeshInstances = [];
+// const tempMeshInstances = [];
```
Comment (willeastcott) [inline src/framework/graphics/picker.js:16]: Delete

File: src/framework/graphics/picker.js [hunk 2]
```diff
@@ -194,73 +190,12 @@ class Picker {
...
-                }
-            }
 
-            DebugGraphics.popGpuMarker(device);
-        });
 
-        DebugHelper.setName(renderPass, `RenderPass-Picker`);
+
+
+        /////////// !!!!!!!! todo  - this should be done once and RT resized when needed
```
Comment (willeastcott) [inline src/framework/graphics/picker.js:197]: Uhm....maybe turn down the volume! 🤣

File: src/scene/renderer/render-pass-cookie-renderer.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+// shader runs for each face, with inViewProj matrix representing a face camera
+const textureCubeBlitFragmentShader = `
+    varying vec2 uv0;
+    uniform samplerCube blitTexture;
+    uniform mat4 invViewProj;
+    void main(void) {
+        vec4 projPos = vec4(uv0 * 2.0 - 1.0, 0.5, 1.0);
+        vec4 worldPos = invViewProj * projPos;
+        gl_FragColor = textureCube(blitTexture, worldPos.xyz);
+    }`;
```
Comment (willeastcott) [inline src/scene/renderer/render-pass-cookie-renderer.js:40]: Personally I wouldn't indent here...

**Review by willeastcott — APPROVED** [review]

File: src/scene/renderer/render-pass-cookie-renderer.js (hunk 3 above)
Comment (mvaligursky) [reply src/scene/renderer/render-pass-cookie-renderer.js:40]: I find it a lot easier to scan the code and see start / end

File: src/framework/graphics/picker.js (hunk 2 above)
Comment (mvaligursky) [reply src/framework/graphics/picker.js]: haha yeah sorry I missed it even when it was so obvious! :)

File: src/scene/renderer/render-pass-cookie-renderer.js (hunk 3 above)
Comment (willeastcott) [reply src/scene/renderer/render-pass-cookie-renderer.js:40]: Is that so?

https://github.com/playcanvas/engine/blob/main/src/platform/graphics/webgl/webgl-graphics-device.js#L46-L84

😉

Comment (mvaligursky) [reply src/scene/renderer/render-pass-cookie-renderer.js:40]: I'll add the tag, thanks. I'll also indent those shaders you mentioned :)
See MorphInstance for a good example.

**Pull request merged**